### PR TITLE
fix evaluation of checksum for empty files

### PR DESCRIPTION
**Pull request by gnusi**

### CODE REVIEW

**Comment by gnusi** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/iresearch_s-pr/169/

File: core/formats/format_utils.cpp [hunk 1]
```diff
@@ -72,35 +69,48 @@ void write_footer(index_output& out) {
...
+  return sizeof(int32_t)*2 +
+         bytes_io<uint64_t>::vsize(format.size()) +
+         format.size();
+}
+
 int32_t check_header(
     index_input& in, 
     const string_ref& req_format,
     int32_t min_ver, int32_t max_ver) {
+  const size_t left = in.length() - in.file_pointer();
```
Comment (Dronplane) [inline core/formats/format_utils.cpp:82]: should we add `assert(in.length() >= in.file_pointer())`

**Review by Dronplane — APPROVED** [review]
LGTM

**Comment by gnusi** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/iresearch_s-pr/172/

**Comment by gnusi** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/iresearch_s-pr/

**Pull request merged**